### PR TITLE
Refactor the keyphraseDistribution research

### PR DESCRIPTION
**Pull request by agnieszkaszuba**
## Summary

This PR can be summarized in the following changelog entry:

* [yoastseo] Creates language-specific keyphraseDistribution researches

## Relevant technical choices:

* The assessment applies different scoring criteria depending on whether the language has function words support or not. That's why the function in the basic research takes a boolean value that tells it whether the language has function words. The value is false in the default research and true in the language-specific researches for languages with function words. However, this means that every language-specific research looks identical (as opposed to other language-specific researches where usually every language is different). Maybe there is a more efficient way to handle this in the future so that there is no need for so many identical files?

## Test instructions

This PR can be tested by following these steps:

*

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes https://yoast.atlassian.net/browse/LINGO-477

### CODE REVIEW

File: packages/yoastseo/src/languageProcessing/languages/de/researches/keyphraseDistribution.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import getFunctionWords from "../config/functionWords";
+const functionWords = getFunctionWords().all;
+
+/**
+ * Determines which portions of the text did not receive a lot of content words from keyphrase and synonyms.
+ *
+ * @inheritDoc keyphraseDistributionResearcher
+ */
+export default function( paper, researcher ) {
+	return keyphraseDistributionResearcher( paper, researcher, functionWords );
```
Comment (omarreiss) [inline packages/yoastseo/src/languageProcessing/languages/de/researches/keyphraseDistribution.js:11]: You pass true or false everywhere else. Here you pass a list. Is this intentional?

File: packages/yoastseo/src/languageProcessing/researches/base/keyphraseDistribution.js [hunk 2]
```diff
@@ -182,10 +178,11 @@ const getSentenceScores = function( sentences, topicFormsInOneArray, locale ) {
  *
  * @param {Paper}       paper       The paper to check the keyphrase distribution for.
  * @param {Researcher}  researcher  The researcher to use for analysis.
+ * @param {boolean}     funcWords   The function words list.
  *
  * @returns {Object} The scores of topic relevance per portion of text and an array of all word forms to highlight.
  */
-const keyphraseDistributionResearcher = function( paper, researcher ) {
+const keyphraseDistributionResearcher = function( paper, researcher, funcWords ) {
```
Comment (omarreiss) [inline packages/yoastseo/src/languageProcessing/researches/base/keyphraseDistribution.js:185]: `funcWords` implies an array. What would be a parameter name that describes the boolean well?

**Review by omarreiss — CHANGES_REQUESTED** [review]
Looking alright. Two things that need fixing.

File: packages/yoastseo/src/languageProcessing/languages/de/researches/keyphraseDistribution.js (hunk 1 above)
Comment (FAMarfuaty) [reply packages/yoastseo/src/languageProcessing/languages/de/researches/keyphraseDistribution.js]: That wasn't intentional. We just fixed it.

File: packages/yoastseo/src/languageProcessing/researches/base/keyphraseDistribution.js (hunk 2 above)
Comment (FAMarfuaty) [reply packages/yoastseo/src/languageProcessing/researches/base/keyphraseDistribution.js]: We changed it to `hasFunctionWords` in the latest commit

**Pull request merged**